### ISSUE TITLE
[backend] Retry run failed.: InternalServerError: Workflow cannot be retried with node ... in Omitted phase: workflow cannot be retried

### ISSUE THREAD

**Issue by skogsbrus** [kubeflow/pipelines#5878]
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)? KF 1.3 Manifests release, on premises

*  KFP version:  1.5.0

*  KFP SDK version: 

```
$ pip list | grep kfp
kfp                      1.6.2    
kfp-pipeline-spec        0.1.7    
kfp-server-api           1.5.0  
```

### Steps to reproduce

1. Run a pipeline with a failing component
2. Retry the run

```
Failed to retry run: 10ff9a33-bc37-4224-914a-8278036a163c with error: "{"error":"Retry run failed.: InternalServerError: Workflow cannot be retried with node pipeline-vlkt4-400844542 in Omitted phase: workflow cannot be retried","code":13,"message":"Retry run failed.: InternalServerError: Workflow cannot be retried with node pipeline-vlkt4-400844542 in Omitted phase: workflow cannot be retried","details":[{"@type":"type.googleapis.com/api.Error","error_message":"Internal Server Error","error_details":"Retry run failed.: InternalServerError: Workflow cannot be retried with node pipeline-vlkt4-400844542 in Omitted phase: workflow cannot be retried"}]}"
```
### Expected result

A successful retry of the run, starting off from the failed component.

### Materials and Reference

I've found this Argo bug.
https://github.com/argoproj/argo-workflows/issues/3524

But unclear if that's the root cause. Retries worked previously when I ran KF 1.1 on premises, but since upgrading to 1.3 they fail with this error.

---

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by capri-xiyue** [comment]
fixed by #5738

**Comment by skogsbrus** [comment]
Ok, thanks!

**Comment by caleb-artifact** [comment]
Any idea when this fix will land in a 1.6 release?
